### PR TITLE
LibCore+flock: Make `Core::System::waitpid` more ergonomic

### PR DESCRIPTION
**Pull request by juniorrantila**

### CODE REVIEW

File: Userland/Utilities/flock.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2020-2022, Andreas Kling <kling@serenityos.org>
```
Comment (awesomekling) [inline Userland/Utilities/flock.cpp:2]: I didn't make this edit. :^)

Comment (juniorrantila) [reply Userland/Utilities/flock.cpp:2]: Went on autopilot and assumed it said the usual "Serenity developers", but you're right. Should I remove the bump then? 👀

Comment (awesomekling) [reply Userland/Utilities/flock.cpp:2]: Yes please.

Comment (awesomekling) [reply Userland/Utilities/flock.cpp:2]: (But feel free to add your own copyright ofc!)

File: Userland/Utilities/flock.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     }
 
     pid_t child_pid = TRY(Core::System::posix_spawnp(arguments.strings[2], nullptr, nullptr, &arguments.argv[2], environ));
-    int status = TRY(Core::System::waitpid(child_pid, &status, 0));
+    auto [_, status] = TRY(Core::System::waitpid(child_pid));
```
Comment (kleinesfilmroellchen) [inline Userland/Utilities/flock.cpp:22]: fancy :^)

**Review by kleinesfilmroellchen — APPROVED** [review]

**Pull request merged**